src/google_civic/core.clj

```diff
@@ -1,39 +1,106 @@
 (ns google-civic.core
   (:require [clj-http.client :as client]
             [cheshire.core :as json]
-            [slingshot.slingshot :refer [try+]]))
+            [clojure.set :as set]))
 
 (def api-root "https://www.googleapis.com/civicinfo/v2")
 
+(def temporary-rate-limit-reasons
+  #{"concurrentLimitExceeded" "rateLimitExceeded"
+    "servingLimitExceeded" "userRateLimitExceeded"})
+
+(defn temporary-rate-limit?
+  "True if the errors in the response only include errors from the
+  temporary-rate-limit-reasons set and the code is 403."
+  [response body]
+  (and (= 403 (:status response))
+       (let [errors (get-in body [:error :errors])
+             reasons (->> errors (map :reason) set)]
+         (and (seq reasons)
+              (set/superset? temporary-rate-limit-reasons
+                             reasons)))))
+
+(defn can-retry?
+  "We can retry if the response code indicates a server error, or
+   if the error is deemed a temporary rate limit."
+  [response body]
+  (cond
+    (client/server-error? response) true
```

Comment:
Minor preference here, I know the examples use the `client` alias I'd prefer to see `http`.